apps/pay/components/receive-amount.tsx

```diff
@@ -29,22 +29,28 @@ export default function ReceiveAmount({
   recipientWalletCurrency: string
 }) {
   const router = useRouter()
+  const searchParams = useSearchParams()
+  const { username } = useParams()
+  const query = extractSearchParams(searchParams)
   const { satsToUsd, usdToSats } = useSatPrice()
-  const { amount, currency } = parseQueryAmount(router.query) // USD or SATs
-  const { display } = parseDisplayCurrency(router.query)
+  const { amount, currency } = parseQueryAmount(query) // USD or SATs
+  const { display } = parseDisplayCurrency(query)
 
   function toggleCurrency() {
     const newCurrency = currency === "SATS" ? "USD" : "SATS"
     const newAmount =
       newCurrency === "SATS" ? Math.round(usdToSats(amount)) : satsToUsd(amount)
 
     router.push(
-      getUpdatedURL(router.query, { currency: newCurrency, amount: newAmount, display }),
```

Comment:
`getUpdatedURL` code is quite smelly to me. I think we may want to either refactor it (can't we just use `URL` and `URLSearchParams`?). having some unit tests otherwise would be useful.

Reply:
This file is not used in app. There are multiple files that are not used in app. We will need a cleanup PR.